frontends/election-manager/src/screens/manual_data_import_precinct_screen.tsx

```diff
@@ -155,80 +315,70 @@ export function ManualDataImportPrecinctScreen(): JSX.Element {
     useParams<ManualDataPrecinctScreenProps>();
   const history = useHistory();
 
+  const existingManualData = fullElectionExternalTallies.get(
+    ExternalTallySourceType.Manual
+  );
+  const ballotType =
+    existingManualData?.votingMethod ?? manualTallyVotingMethod;
+  const existingTalliesByPrecinct: Dictionary<TempExternalTally> | undefined =
+    existingManualData?.resultsByCategory.get(TallyCategory.Precinct);
+
   const currentPrecinct = election.precincts.find(
     (p) => p.id === currentPrecinctId
   );
-  if (currentPrecinct === undefined) {
-    return (
-      <Prose>
-        Error: Could not find precinct {currentPrecinctId}.{' '}
-        <LinkButton to={routerPaths.manualDataImport}>Back to Index</LinkButton>
-      </Prose>
-    );
-  }
-  const existingManualData = fullElectionExternalTallies.get(
-    ExternalTallySourceType.Manual
+  const existingPrecinctTally = existingTalliesByPrecinct
+    ? existingTalliesByPrecinct[currentPrecinctId]
+    : undefined;
+
+  const [talliesByPrecinct, setTalliesByPrecinct] = useState(
+    existingTalliesByPrecinct
   );
-  const existingTalliesByPrecinct = existingManualData?.resultsByCategory.get(
-    TallyCategory.Precinct
+  const [currentPrecinctTally, setCurrentPrecinctTally] = useState(
+    existingPrecinctTally
   );
-  const talliesByPrecinct: Dictionary<TempExternalTally> =
-    existingTalliesByPrecinct ?? getEmptyExternalTalliesByPrecinct(election);
-
-  const initialPrecinctTally =
-    talliesByPrecinct[currentPrecinctId] ?? getEmptyExternalTally();
-  const [currentPrecinctTally, setCurrentPrecinctTally] =
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    useState(initialPrecinctTally);
-
-  const ballotType = existingManualData?.votingMethod || VotingMethod.Precinct;
-
-  // Convert internal structure of contest data that allows for empty strings, to the regular
-  // type by mapping any empty string values to zeros.
-  function convertContestTallies(
-    contestTallies: Dictionary<TempContestTally>
-  ): Dictionary<ContestTally> {
-    const convertedContestTallies: Dictionary<ContestTally> = {};
-    for (const contestId of Object.keys(contestTallies)) {
-      const contestTally = contestTallies[contestId];
-      assert(contestTally);
-      const convertedOptionTallies: Dictionary<ContestOptionTally> = {};
-      for (const optionId of Object.keys(contestTally.tallies)) {
-        const optionTally = contestTally.tallies[optionId];
-        assert(optionTally);
-        convertedOptionTallies[optionId] = {
-          ...optionTally,
-          tally: optionTally.tally === '' ? 0 : optionTally.tally,
-        };
+
+  const writeInSummaryQuery = useWriteInSummaryQuery({ status: 'adjudicated' });
+  // Get empty tallies with previously adjudicated candidate names, only
+  // when none already exist at initial page load
+  useEffect(() => {
+    if (talliesByPrecinct) return;
+    if (!writeInSummaryQuery.data) return;
```

Comment:
I forget exactly the right thing to check for but we were seeing a bug with the tally reports where this would get fetched twice right away and the first time would be stale data, so I would just make sure to test adding a new adjudication value, and then going to this screen first thing and making sure that value shows up.

Reply:
I had the same issue in development, and that's why I added the `writeInSummaryQuery.isFetchedAfterMount` check. I agree it would be good to bake into the test for the screen.

Reply:
After thinking about it a bit, I think this is covered by the integration test, actually, so I will leave it for now.

Reply:
Edited end-to-end test to check this.